File: leetcode/typescript/2009_minimum_number_of_operations_to_make_array_continuous.ts
function minOperations(nums: number[]): number {
    const n = nums.length;
    const sortedNums = Array.from(new Set(nums)).sort((a, b) => a - b);
    let res = n;

    for (const [i, l] of sortedNums.entries()) {
        const r = l + n - 1;
        let j = i;
        bisectRight(sortedNums, r);
        const count = j - i;
        res = Math.min(res, n - count);
    }
    return res;
};

function bisectRight(nums: number[], target: number): number {
    let left = 0;
    let right = nums.length;

    while (left < right) {
        const mid = Math.floor((left + right) / 2);

        if (nums[mid] > target) {
            right = mid;
        } else {
            left = mid + 1;
        }
    }

    return left;
}

File: leetcode/python/2009_minimum_number_of_operations_to_make_array_continuous.py
from bisect import bisect_right


class Solution:
    def minOperations(self, nums: list[int]) -> int:
        n = len(nums)
        sorted_nums = sorted(set(nums))
        res = n
        
        for i, l in enumerate(sorted_nums):
            r = l + n - 1
            j = bisect_right(sorted_nums, r)
            count = j - i
            res = min(res, n - count)
        
        return res